pyscriptjs/src/components/pyrepl.ts

```diff
@@ -51,6 +51,42 @@ export function make_PyRepl(interpreter: InterpreterClient, app: PyScriptApp) {
             logger.debug(`element ${this.id} successfully connected`);
         }
 
+        get src() {
+            return this.getAttribute('src');
+        }
+
+        set src(value) {
+            this.setAttribute('src', value);
+        }
+
+        attributeChangedCallback(name: string, oldVal: string, newVal: string) {
+            if (name === 'src' && newVal !== oldVal) {
+                void this.loadReplSrc();
+            }
+        }
+
+        /**
+         * Fetch url from src attribute of py-repl tags and
+         * preload the code from fetch response into the Corresponding py-repl tag,
+         * but please note that they will not be pre-run unless you click the runbotton.
+         */
+        async loadReplSrc() {
+            const response = await fetch(this.src);
```

Comment:
Small thing, we should use `robustFetch` here which you can find in`fetch.ts` since this function will do error handling and raise UserError which will create a banner for us 😃 

https://github.com/pyscript/pyscript/blob/main/pyscriptjs/src/fetch.ts#L12

Reply:
good catch ... just for my personal understanding, do you think we should do that any time we think we want fetch instead?

Reply:
I think so, I tried to handle status code and showing the proper message then return the response. But you have way more experience than me, let me know if you see something terrible there 😆